### PR TITLE
Fix compile errors with Qt 5.3.2 and Boost 1.55.0

### PR DESCRIPTION
**Pull request by ryanofsky**
Both errors were introduced in #9974 and reported by @paveljanik

### CODE REVIEW

**Comment by paveljanik** [comment]
Going to test this. Thanks!

**Comment by paveljanik** [comment]
Compiles OK on Debian Jessie and openSUSE (gcc 4.8.5, 5.3.1 and 6.2.1).

ACK
https://github.com/bitcoin/bitcoin/pull/10039/commits/b5bec4e330fc7201d989663b4dbc6a1e620dd0f9

**Comment by pstratem** [comment]
This chance breaks compiling on debian 8.7 with qt4 ( 4:4.8.6+git64-g5dc8b2b+dfsg-3+deb8u1 ).

I suspect one version works in qt4 and the other in qt5.

**Comment by pstratem** [comment]
@ryanofsky ping

**Comment by jonasschnelli** [comment]
Maybe it's time to think about dropping support to compile against Qt4? We already dropped support during runtime...

**Comment by laanwj** [comment]
> Maybe it's time to think about dropping support to compile against Qt4? We already dropped support during runtime...

The issue for discussion that is #8263

File: src/qt/test/wallettests.cpp [hunk 1]
```diff
@@ -22,7 +22,9 @@ namespace
 //! Press "Yes" button in modal send confirmation dialog.
 void ConfirmSend()
 {
-    QTimer::singleShot(0, Qt::PreciseTimer, []() {
+    QTimer* timer = new QTimer;
+    timer->setSingleShot(true);
+    QObject::connect(timer, &QTimer::timeout, []() {
```
Comment (gmaxwell) [inline src/qt/test/wallettests.cpp]: This fails to compile for me on debian testing with qt/test/wallettests.cpp:36:6: error: no matching function for call to ‘QObject::connect(QTimer*&, void (QTimer::*)(), 
                  {anonymous}::ConfirmSend()::<lambda()>)’ });

The compile is successful if I revert this commit.

Comment (ryanofsky) [inline src/qt/test/wallettests.cpp]: #10098 fixes this

**Pull request merged**